src/components/tree-views-container.tsx

```diff
@@ -411,11 +432,9 @@ const TreeViewsContainer = () => {
         [updateCurrentChildren, updateMapData]
     );
 
-    // add ghost studies or ghost cases as soon as possible (uploadingElements)
+    // add ghost studies or ghost cases as soon as possible (uploadingElements) and clean them if necessary
     useEffect(() => {
-        if (Object.values(uploadingElements).length > 0) {
-            dispatch(setCurrentChildren(mergeCurrentAndUploading(currentChildrenRef.current ?? [])));
-        }
+        dispatch(setCurrentChildren(mergeCurrentAndUploading(currentChildrenRef.current ?? [])));
```

Comment:
ok there to not check the size of `uploadingElements` because we need to clean current if necessary